api/src/db-connection.ts

```diff
@@ -389,6 +399,216 @@ export class Connection {
     return entities;
   }
 
+  /**
+   * Default implementation in `getEntitiesForPaper` naively retrieves all entities,
+   * which includes quadratic data duplication across multiple instances of the same symbol.
+   * This variant method is meant as a placeholder bridge until the underlying DB schema and
+   * extraction write layer is changed to be non-duplicative.
+   * Leaving the underlying data unchanged in the DB, we selectively exclude the entity data
+   * types known to be redundant. These are retrieved later and added into lookup tables.
+   */
+  async getDedupedEntitiesForPaper(paperSelector: PaperSelector, entityTypes: EntityType[], version?: number) {
+    if (version === undefined) {
+      try {
+        let latestVersion = await this.getLatestPaperDataVersion(paperSelector);
+        if (latestVersion === null) {
+          return [];
+        }
+        version = latestVersion;
+      } catch (e) {
+        console.log("Error fetching latest data version number:", e);
```

Comment:
I feel like we should probably be throwing `e` up the stack here so it can be caught in the API and we actually 500 if this happens.

Reply:
Hm this is a straight copy pasta from the original method. Would it help for me to delineate what's new in this version inline?